### PR TITLE
Made an MLP baseline; Moved _get_data_loader and _get_optimizer to the base class to avoid code duplication

### LINKED ISSUE #145: DNA Ablation test: MLP metafeatures to pipeline score (no pipeline information)

### PR DESCRIPTION
**Pull request by erikhuck**
closes #145

### CODE REVIEW

**Comment by bjschoenfeld** [comment]
@erikhuck can you update this with develop and incorporate the `loss_function_name` changes?

File: dna/models/baselines.py [hunk 1]
```diff
@@ -6,8 +6,11 @@
 import pandas as pd
 from sklearn import linear_model
 from sklearn.ensemble import RandomForestRegressor
+import torch
+from .torch_modules.mlp import MLP
 
-from .base_models import RankModelBase, RegressionModelBase, SklearnBase, SubsetModelBase
+from .base_models import RankModelBase, RegressionModelBase, SklearnBase, SubsetModelBase,\
```
Comment (bjschoenfeld) [inline dna/models/baselines.py]: A common practice for multi-line imports is to is parentheses:
```
from module import (
    a, b, c, d, e
    f, g, h, i, j
)
```
It's not that important though.

Comment (erikhuck) [reply dna/models/baselines.py]: Thanks for letting me know.

**Comment by bjschoenfeld** [comment]
This is failing on the rank problem:

```
.../d3m-dynamic-neural-architecture/env/lib/python3.6/site-packages/numpy/lib/function_base.py:2530: RuntimeWarning: invalid value encountered in true_divide
  c /= stddev[:, None]
...
.../d3m-dynamic-neural-architecture/env/lib/python3.6/site-packages/scipy/stats/_distn_infrastructure.py:901: RuntimeWarning: invalid value encountered in greater
  return (a < x) & (x < b)
```

There are a couple more warnings. The scores are invalid:

```
"per_dataset_scores": {
    "LL0_1100_popularkids_dataset": {
        "spearman_correlation": {
            "correlation_coefficient": NaN,
            "p_value": NaN
        }
    }
}
...
"mean_scores": {
    "spearman_correlation": {
        "mean": NaN,
        "std_dev": NaN,
        "mean_p_value": NaN,
        "std_dev_p_value": NaN
    }
}
```

**Pull request merged**